documentation/general/deployment.md

```diff
@@ -163,8 +163,11 @@ realise that the deployed code is not behaving as we expected. In these cases it
 may be necessary to force an environment to be deployed to a specific version,
 usually the previous version of the application.
 
-The same staging and production deployment workflows are used directly to
-rollback any environment for any service. Only members of the
+The same staging and production deployment workflows in the
+WordPress/openverse-infrastructure repository are used directly to rollback any
+environment for any service. **Do not attempt to perform a rollback using the
```

Comment:
Again, is this accurate? My understanding is that rollbacks should be done _only_ by running the deployment workflow from the infra repo, with the latest stable version.

Would triggering a new release at https://github.com/WordPress/openverse/releases/new with the stable version result in creating duplicate tags?

Reply:
Yes, that is correct. Really we can frame it like this: to deploy a specific, already released version, run the deployment workflow in the infrastructure repository with the version tag to deploy. This applies to _any_ (literally, all possible cases) scenario where we need to trigger a dpeloyment to an existing release version, whether a rollback or re-deployment (e.g., to get new environment variables) or whatever.

Publishing a release _creates the release_ and the release app workflow is just an incidental part of publishing (it would be good if we just renamed it to "Tag stable image and publish changelog" (TSIAPC :cold_sweat:) or something, as it is _not_ in any way independent of publishing). Publishing is only for creating new a new release off of `HEAD` on `main` and has no other purpose.

The _deployment_ workflows are general purpose deployment workflows utilised by the "TSIAPC" workflow (release-app) as an incidental part of publishing a release, but not a _strictly necessary_ one. We could, for example, remove automated deployments for the API and frontend from the "TSIAPC" workflow (release-app), to completely separate the ideas and prevent any possible confusion between the two. _Publishing is publishing_. _Deploying is deploying_. There is no necessary crossover between the two, which is why they are distinct workflows with no direct relationship to each other, aside from that publishing _happens_ to trigger the deployment workflows.

This might sound pedantic, but it's really important to completely separate these two workflows conceptually, and for everyone creating releases and/or deploying the application _in any capacity_ understands the separation between them. Publishing _triggers_ deployment, but is not _part of_ deployment. Deployment is also not part of publishing a release, it just happens to occur after publishing a release because that's basically why we publish releases, to immediately deploy to production.

It really might be best to just separate the two and not have publishing trigger a deployment, to emphasise the separation of the two... but, as it is literally what we want to do every time we publish a release, I think we would quickly find it tedious that the frontend and API no longer automatically deploy to production when we create a new release.

I'm also thinking there is a confusion of the GitHub terms and what would be the ideal, precise words to use: Publishing implies making the thing public, which in our case really means deploying it to production (because we run apps, not libraries). But, ideally, the process would be called "Create a stable version with a changelog, then deploy it", which emphasises the two completely separate parts. It's just that GitHub call's it "publishing a release", wihch overloads a whole lot of ideas that for us are actually separate. If we think of publishing as a combination of creating the release and deployment, it's still kind of confusing.

Anyway, the concepts are currently muddled, and the words GitHub uses and that we've titled our workflow files with are not helping maintain the separation. If removing the automated deployments from the workflow triggered by publishing a release in GitHub would help make that clearer, then it's probably worth the tediousness, as mistakes here are potentially more confusing than the tediousness.